frame/transaction-payment/src/lib.rs

```diff
@@ -51,10 +51,10 @@ use sp_runtime::{
 	},
 	traits::{
 		Saturating, SignedExtension, SaturatedConversion, Convert, Dispatchable,
-		DispatchInfoOf, PostDispatchInfoOf,
+		DispatchInfoOf, PostDispatchInfoOf
 	},
 };
-use pallet_transaction_payment_rpc_runtime_api::RuntimeDispatchInfo;
+use pallet_transaction_payment_rpc_runtime_api::{FeeDetails, InclusionFee, RuntimeDispatchInfo};
```

Comment:
I strongly advise for these two structs to be defined in the root of this pallet, not merely in the rpc api. The reason is that `FeeDetails` is dependent on how this pallet is working now (i.e. a notion of base fee, tip, weight fee, etc) and the correct flow of imports, in my opinion, is for the rpc crate to use these structs from the main crate, not the other way around.

Reply:
I agree, but we need to also move `RuntimeDispatchInfo` to this pallet, otherwise, we'll have the cyclic dependency problem. Or we can add these structs into `primitives/transaction-payment`? What's your opinion?

Reply:
I would move both to the root pallet.